Display images on Adoptions page

Dindihub opened the pull request:
Display images on Adoption's page to finish setting up the page.

Please review @chalin @lukpueh

Dindihub committed:
Display images on adoptions page

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub committed:
fix format issues

Signed-off-by: sandra <sadindi03@gmail.com>

chalin reviewed (commented):
This PR will need some rework following the merge of #19. See

- #16 

for a description of what needs to be done.

Note: instead of putting Adoptions images under `static`, put them under `/content/en/community/adoptions/img`.

Dindihub commented:
> This PR will need some rework following the merge of #19. See
> 
> * [Community section adjustments #16](https://github.com/Dindihub/docsy-tuf.io/issues/16)
> 
> for a description of what needs to be done.
> 
> Note: instead of putting Adoptions images under `static`, put them under `/content/en/community/adoptions/img`.

@chalin Okay. I was thinking of opening a separate PR for that but this works too.

Dindihub committed:
move adoptions to community section

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub commented:
> > This PR will need some rework following the merge of #19. See
> > 
> > * [Community section adjustments #16](https://github.com/Dindihub/docsy-tuf.io/issues/16)
> > 
> > for a description of what needs to be done.
> > Note: instead of putting Adoptions images under `static`, put them under `/content/en/community/adoptions/img`.
> 
> @chalin Okay. I was thinking of opening a separate PR for that but this works too.

@chalin  may I ask why you want the images moved from static to community? It really complicated the code, it wouldn't run. I had to take it back to static :)  Not sure what I'm doing wrong. 

With the images in static it works just fine.

I tried changing the imgURL on the adoptions.html from   `{{ $imgUrl := printf "img/logos/adoptions/%s_logo.%s" $img $type | relURL }}` to   `{{ $imgUrl := printf "en/community/adoptions/img/%s_logo.%s" $img $type | relURL }}` but it wouldn't work.

chalin commented:
In #17, I shared with you have to rebase. **Please rebase this PR**.

Also, can you tell me what output you get when you run `git remote -v`? If you see `upstream`, then remove it by running the following command: `git remote remove upstream`.  You won't be contributing to the upstream project, and it just causes headaches otherwise.

chalin commented:
> @chalin may I ask why you want the images moved from static to community?

For cohesion: the images are _only_ used in the adoptions section.

> I tried changing the imgURL on the adoptions.html from `{{ $imgUrl := printf "img/logos/adoptions/%s_logo.%s" $img $type | relURL }}` to `{{ $imgUrl := printf "en/community/adoptions/img/%s_logo.%s" $img $type | relURL }}` but it wouldn't work.

After you've moved the files, the path to the images should simply be `/img/%s_logo.%s` because the path will be local to the folder. The full path you gave above should work too, as long as you drop the `en/` prefix (if you look at the preview, there's no `en/` in the URL path). Does that help?

Dindihub commented:
> In #17, I shared with you have to rebase.
> 
> **Please rebase this PR**. You'll need to resolve the conflicts that arise.
> 
> Also, can you tell me what output you get when you run `git remote -v`? If you see `upstream`, then remove it by running the following command: `git remote remove upstream`. You won't be contributing to the upstream project, and it just causes headaches otherwise.

Rebased and removed upstream. I realized my mistake was rebasing on local but forgetting to --force push. I think it's solved now.

Dindihub committed:
move adoptions logos to community

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub commented:
> > @chalin may I ask why you want the images moved from static to community?
> 
> For improved cohesion: the images are _only_ used in the adoptions section.
> 
> > I tried changing the imgURL on the adoptions.html from `{{ $imgUrl := printf "img/logos/adoptions/%s_logo.%s" $img $type | relURL }}` to `{{ $imgUrl := printf "en/community/adoptions/img/%s_logo.%s" $img $type | relURL }}` but it wouldn't work.
> 
> After you've moved the files, the path to the images should simply be `/img/%s_logo.%s` because the path will be local to the folder. The full path you gave above should work too, as long as you drop the `en/` prefix (if you look at the preview, there's no `en/` in the URL path). Does that help?

It works now. Thanks

Dindihub committed:
fix links

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub committed:
fix format

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub committed:
fix file names

Signed-off-by: sandra <sadindi03@gmail.com>

Dindihub commented:
Please review again @chalin . Thanks

chalin commented on content/en/community/adoptions/adoptions.md:
Rename this to `index.md` (note the lack of underscore); otherwise we end up with this stutter in the path:

> ![image](https://github.com/user-attachments/assets/ff757367-c973-42d8-a546-c01a0d54b5ff)

chalin commented on content/en/community/adoptions/img/logos/agl-logo.png:
Rename the folder `content/en/community/adoptions/img/logos` to just `content/en/community/adoptions/img`. We don't need to put the images in a logos subfolder.

chalin commented on content/en/_index.md:
```diff
@@ -39,7 +39,7 @@ Discover how TUF secures update systems
 {{% /blocks/feature %}}
 
-{{% blocks/feature icon="fa-solid fa-gear" title="Production Ready" url="docs/adoptions/" %}}
+{{% blocks/feature icon="fa-solid fa-gear" title="Production Ready" url="community/" %}}
```
```suggestion
{{% blocks/feature icon="fa-solid fa-gear" title="Production Ready" url="/community/adoptions/" %}}
```

chalin commented on content/en/docs/get-started/adopter.md:
```diff
@@ -8,9 +8,9 @@ TUF provides a framework for integration of the
-While some [adoptions](docs/adoptions) integrate TUF by implementing the
-framework from scratch, others start from either a TUF implementation or from a
-TUF system.
+While some [adoptions](/community) integrate TUF by implementing the framework
```
```suggestion
While some [adoptions](/community/adoptions/) integrate TUF by implementing the framework
```

chalin commented on content/en/docs/get-started/adopter.md:
This page shouldn't exist under `/get-started/`, but we'll address this separately. I'll file an issue.

chalin commented on layouts/shortcodes/adoptions.html:
```diff
@@ -0,0 +1,19 @@
+    {{ $img    := .tag | lower }}
+    {{ $type   := .img_type | default "png" }}
+    {{ $alt    := .alt | default .tag }}
+    {{ $imgUrl := printf "community/adoptions/img/logos/%s_logo.%s" $img $type | relURL }}
```
Note that the compound file names use hyphens not underscores. Also, we can use a relative path, as I'd mentioned in the previous round of edits:

```suggestion
    {{ $imgUrl := printf "img/%s-logo.%s" $img $type }}
```

chalin reviewed (commented):
Getting closer, but the images are still not rendering. Run `npm test` locally and you'll see the invalid links.

For the sake of making progress on this PR, I'll address the issues in a followup commit over this PR.

chalin committed:
Fix location and paths to logos

chalin committed:
Drop `-logo` file name base suffix

chalin committed:
Link directly to Adoptions page

chalin committed:
Link directly to Adoptions page

chalin committed:
Fix format

chalin committed:
Final prose tweak

chalin reviewed (approved):
LGTM! All tests are passing (locally and via GH actions), and the images are rendering correctly in the Adoptions page.

chalin commented:
Merging. @lukpueh if you have any comments we can address them via a followup PR.
